test/Path/Gen.hs

```diff
@@ -76,3 +76,6 @@ genFilePath = listOf genPathyChar
 
 genPathyChar :: Gen Char
 genPathyChar = frequency [(2, arbitrary), (1, elements "./\\")]
+
+shrinkPath :: Path a b -> [Path a b]
+shrinkPath (Path s) = Path <$> shrink s
```

Comment:
For comparison, [here](https://github.com/haskell/filepath/blob/141cddb606fd6f6a60f730fed3d57502b93c14ae/tests/TestUtil.hs#L45)'s an interesting shrinking function for valid `FilePath`s.